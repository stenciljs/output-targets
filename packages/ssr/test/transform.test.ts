import fs from 'node:fs/promises';
import path from 'node:path';
import url from 'node:url';

import { describe, it, expect } from 'vitest';
import esbuild from 'esbuild';

import { transform } from '../src/transform.js';

function removeFileNameLines (code: string) {
  return code.replace(/^\s*fileName:.*,\n/gm, '');
}

const __dirname = path.dirname(url.fileURLToPath(import.meta.url));
const templatePath = path.resolve(__dirname, '__fixtures__', 'app.tsx')
const template = await fs.readFile(templatePath, 'utf-8');
const code = removeFileNameLines((await esbuild.transform(template, {
  loader: 'tsx',
  jsx: 'automatic', // Use React 17+ JSX transform
  jsxDev: true,
  sourcefile: templatePath,
})).code);

describe('transform', () => {
  it('should transform the code', async () => {
    const transformedCode = await transform(code, templatePath, {
      from: 'component-library-react',
      module: import('component-library-react'),
      hydrateModule: import('component-library/hydrate'),
      serializeShadowRoot: 'scoped',
    })
    if (!transformedCode) {
      throw new Error('No transformed code')
    }
    expect(removeFileNameLines(transformedCode)).toMatchSnapshot()
  })
})
