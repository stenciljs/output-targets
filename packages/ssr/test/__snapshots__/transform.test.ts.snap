// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`transform > should transform the code 1`] = `
"import Input from "../components/Input";

import { MyComponent, MyCounter, MyList, MyListItem } from "component-library-react";

import { jsxDEV } from "react/jsx-dev-runtime";


const MyComponent$0 = ({ children, ...props }) => {
  return /* @__PURE__ */ jsxDEV("div", { style: { display: "contents" }, suppressHydrationWarning: true, children: [
    /* @__PURE__ */ jsxDEV("style", { children: \`
        /*!@:host*/.sc-my-component-h{display:block;color:green}
      \` }, void 0, false, {
      lineNumber: 4,
      columnNumber: 7
    }),
    /* @__PURE__ */ jsxDEV("my-component", { class: "hydrated sc-my-component-h", first: "John", "kids-names": "serialized:eyJ0eXBlIjoiYXJyYXkiLCJ2YWx1ZSI6W3sidHlwZSI6InN0cmluZyIsInZhbHVlIjoiSm9obiJ9LHsidHlwZSI6InN0cmluZyIsInZhbHVlIjoiSmFuZSJ9LHsidHlwZSI6InN0cmluZyIsInZhbHVlIjoiSmltIn1dfQ==", last: "Doe", middle: "William", "s-id": "1", ...props, suppressHydrationWarning: true, dangerouslySetInnerHTML: { __html: \`<!--r.1-->
  <div c-id="1.0.0.0" class="sc-my-component">
    <!--t.1.1.1.0-->
    Hello, World! I'm John undefined Doe
  </div>\` } }, void 0, false, {
      lineNumber: 8,
      columnNumber: 7
    })
  ] }, void 0, true, {
    lineNumber: 3,
    columnNumber: 13
  });
};
const MyComponent$1 = ({ children, ...props }) => {
  return /* @__PURE__ */ jsxDEV("div", { style: { display: "contents" }, suppressHydrationWarning: true, children: [
    /* @__PURE__ */ jsxDEV("style", { children: \`
        /*!@:host*/.sc-my-component-h{display:block;color:green}
      \` }, void 0, false, {
      lineNumber: 19,
      columnNumber: 7
    }),
    /* @__PURE__ */ jsxDEV("my-component", { class: "hydrated sc-my-component-h", "s-id": "2", ...props, suppressHydrationWarning: true, dangerouslySetInnerHTML: { __html: \`<!--r.2-->
  <div c-id="2.0.0.0" class="sc-my-component">
    <!--t.2.1.1.0-->
    Hello, World! I'm undefined undefined undefined
  </div>\` } }, void 0, false, {
      lineNumber: 23,
      columnNumber: 7
    })
  ] }, void 0, true, {
    lineNumber: 18,
    columnNumber: 13
  });
};
const MyList$2 = ({ children, ...props }) => {
  return /* @__PURE__ */ jsxDEV("div", { style: { display: "contents" }, suppressHydrationWarning: true, children: [
    /* @__PURE__ */ jsxDEV("style", { children: \`
        
      \` }, void 0, false, {
      lineNumber: 34,
      columnNumber: 7
    }),
    /* @__PURE__ */ jsxDEV("my-list", { class: "hydrated sc-my-list-h", "s-id": "3", ...props, suppressHydrationWarning: true, dangerouslySetInnerHTML: { __html: \`<!--r.3-->
  <ul c-id="3.0.0.0" class="sc-my-list sc-my-list-s">
    <!--s.3.1.1.0.-->
  </ul>\` } }, void 0, false, {
      lineNumber: 38,
      columnNumber: 7
    })
  ] }, void 0, true, {
    lineNumber: 33,
    columnNumber: 13
  });
};
const MyListItem$3 = ({ children, ...props }) => {
  return /* @__PURE__ */ jsxDEV("div", { style: { display: "contents" }, suppressHydrationWarning: true, children: [
    /* @__PURE__ */ jsxDEV("style", { children: \`
        
      \` }, void 0, false, {
      lineNumber: 48,
      columnNumber: 7
    }),
    /* @__PURE__ */ jsxDEV("my-list-item", { class: "hydrated sc-my-list-item-h", "s-id": "4", ...props, suppressHydrationWarning: true, dangerouslySetInnerHTML: { __html: \`<!--r.4-->
  <!--o.0.1.-->
  <li c-id="4.0.0.0" class="sc-my-list-item sc-my-list-item-s">
    <!--s.4.1.1.0.-->
    <!--t.0.1-->
    Item 1
  </li>\` } }, void 0, false, {
      lineNumber: 52,
      columnNumber: 7
    })
  ] }, void 0, true, {
    lineNumber: 47,
    columnNumber: 13
  });
};
const MyListItem$4 = ({ children, ...props }) => {
  return /* @__PURE__ */ jsxDEV("div", { style: { display: "contents" }, suppressHydrationWarning: true, children: [
    /* @__PURE__ */ jsxDEV("style", { children: \`
        
      \` }, void 0, false, {
      lineNumber: 65,
      columnNumber: 7
    }),
    /* @__PURE__ */ jsxDEV("my-list-item", { class: "hydrated sc-my-list-item-h", "s-id": "5", ...props, suppressHydrationWarning: true, dangerouslySetInnerHTML: { __html: \`<!--r.5-->
  <!--o.0.2.-->
  <li c-id="5.0.0.0" class="sc-my-list-item sc-my-list-item-s">
    <!--s.5.1.1.0.-->
    <!--t.0.2-->
    Item 2
  </li>\` } }, void 0, false, {
      lineNumber: 69,
      columnNumber: 7
    })
  ] }, void 0, true, {
    lineNumber: 64,
    columnNumber: 13
  });
};
const MyListItem$5 = ({ children, ...props }) => {
  return /* @__PURE__ */ jsxDEV("div", { style: { display: "contents" }, suppressHydrationWarning: true, children: [
    /* @__PURE__ */ jsxDEV("style", { children: \`
        
      \` }, void 0, false, {
      lineNumber: 82,
      columnNumber: 7
    }),
    /* @__PURE__ */ jsxDEV("my-list-item", { class: "hydrated sc-my-list-item-h", "s-id": "6", ...props, suppressHydrationWarning: true, dangerouslySetInnerHTML: { __html: \`<!--r.6-->
  <!--o.0.3.-->
  <li c-id="6.0.0.0" class="sc-my-list-item sc-my-list-item-s">
    <!--s.6.1.1.0.-->
    <!--t.0.3-->
    Item 3
  </li>\` } }, void 0, false, {
      lineNumber: 86,
      columnNumber: 7
    })
  ] }, void 0, true, {
    lineNumber: 81,
    columnNumber: 13
  });
};
const MyCounter$6 = ({ children, ...props }) => {
  return /* @__PURE__ */ jsxDEV("div", { style: { display: "contents" }, suppressHydrationWarning: true, children: [
    /* @__PURE__ */ jsxDEV("style", { children: \`
        /*!@:host button*/.sc-my-counter-h button.sc-my-counter{padding:5px 10px;font-size:20px;margin:0 10px}/*!@:host span*/.sc-my-counter-h span.sc-my-counter{font-size:20px;margin:0 10px}
      \` }, void 0, false, {
      lineNumber: 99,
      columnNumber: 7
    }),
    /* @__PURE__ */ jsxDEV("my-counter", { class: "hydrated sc-my-counter-h", "s-id": "7", ...props, suppressHydrationWarning: true, dangerouslySetInnerHTML: { __html: \`<!--r.7-->
  <div c-id="7.0.0.0" class="sc-my-counter">
    <button c-id="7.1.1.0" class="sc-my-counter">
      <!--t.7.2.2.0-->
      -
    </button>
    <span c-id="7.3.1.1" class="sc-my-counter">
      <!--t.7.4.2.0-->
      0
    </span>
    <button c-id="7.5.1.2" class="sc-my-counter">
      <!--t.7.6.2.0-->
      +
    </button>
  </div>\` } }, void 0, false, {
      lineNumber: 103,
      columnNumber: 7
    })
  ] }, void 0, true, {
    lineNumber: 98,
    columnNumber: 13
  });
};


export const meta = () => {
  return [
    { title: "New Remix App" },
    { name: "description", content: "Welcome to Remix!" }
  ];
};
export default function Index() {
  return (
    /* @__PURE__ */ jsxDEV("div", { className: "flex h-screen items-center justify-center", children: /* @__PURE__ */ jsxDEV("div", { className: "flex flex-col items-center gap-16", children: [
      /* @__PURE__ */ jsxDEV("header", { className: "flex flex-col items-center gap-9", children: [
        /* @__PURE__ */ jsxDEV("h1", { className: "leading text-2xl font-bold text-gray-800 dark:text-gray-100", children: [
          "Welcome to ",
          /* @__PURE__ */ jsxDEV("span", { className: "sr-only", children: "Remix" }, void 0, false, {
            lineNumber: 18,
            columnNumber: 24
          }, this)
        ] }, void 0, true, {
          lineNumber: 17,
          columnNumber: 11
        }, this),
        /* @__PURE__ */ jsxDEV("div", { className: "h-[144px] w-[434px]", children: [
          /* @__PURE__ */ jsxDEV(
            "img",
            {
              src: "/logo-light.png",
              alt: "Remix",
              className: "block w-full dark:hidden"
            },
            void 0,
            false,
            {
              lineNumber: 21,
              columnNumber: 13
            },
            this
          ),
          /* @__PURE__ */ jsxDEV(
            "img",
            {
              src: "/logo-dark.png",
              alt: "Remix",
              className: "hidden w-full dark:block"
            },
            void 0,
            false,
            {
              lineNumber: 26,
              columnNumber: 13
            },
            this
          )
        ] }, void 0, true, {
          lineNumber: 20,
          columnNumber: 11
        }, this)
      ] }, void 0, true, {
        lineNumber: 16,
        columnNumber: 9
      }, this),
      /* @__PURE__ */ jsxDEV("nav", { className: "flex flex-col items-center justify-center gap-4 rounded-3xl border border-gray-200 p-6 dark:border-gray-700", children: [
        /* @__PURE__ */ jsxDEV("p", { className: "leading-6 text-gray-700 dark:text-gray-200", children: "What's next?" }, void 0, false, {
          lineNumber: 34,
          columnNumber: 11
        }, this),
        /* @__PURE__ */ jsxDEV(MyComponent$0, {
      first: "John",
      middle: "William",
      last: "Doe",
      kidsNames: ["John", "Jane", "Jim"],
      suppressHydrationWarning: true
    }, void 0, false, {
          lineNumber: 37,
          columnNumber: 11
        }, this),
        /* @__PURE__ */ jsxDEV(MyComponent$1, {
      suppressHydrationWarning: true
    }, void 0, false, {
          lineNumber: 38,
          columnNumber: 11
        }, this),
        /* @__PURE__ */ jsxDEV(MyList$2, {
      children: [
            /* @__PURE__ */ jsxDEV(MyListItem$3, {
        children: "Item 1",
        suppressHydrationWarning: true
      }, void 0, false, {
              lineNumber: 40,
              columnNumber: 13
            }, this),
            /* @__PURE__ */ jsxDEV(MyListItem$4, {
        children: "Item 2",
        suppressHydrationWarning: true
      }, void 0, false, {
              lineNumber: 41,
              columnNumber: 13
            }, this),
            /* @__PURE__ */ jsxDEV(MyListItem$5, {
        children: "Item 3",
        suppressHydrationWarning: true
      }, void 0, false, {
              lineNumber: 42,
              columnNumber: 13
            }, this)
          ],

      suppressHydrationWarning: true
    }, void 0, true, {
          lineNumber: 39,
          columnNumber: 11
        }, this),
        /* @__PURE__ */ jsxDEV(Input, {}, void 0, false, {
          lineNumber: 44,
          columnNumber: 11
        }, this),
        /* @__PURE__ */ jsxDEV(MyCounter$6, {
      suppressHydrationWarning: true
    }, void 0, false, {
          lineNumber: 45,
          columnNumber: 11
        }, this),
        /* @__PURE__ */ jsxDEV("ul", { children: resources.map(({ href, text, icon }) => /* @__PURE__ */ jsxDEV("li", { children: /* @__PURE__ */ jsxDEV(
          "a",
          {
            className: "group flex items-center gap-3 self-stretch p-3 leading-normal text-blue-700 hover:underline dark:text-blue-500",
            href,
            target: "_blank",
            rel: "noreferrer",
            children: [
              icon,
              text
            ]
          },
          void 0,
          true,
          {
            lineNumber: 49,
            columnNumber: 17
          },
          this
        ) }, href, false, {
          lineNumber: 48,
          columnNumber: 15
        }, this)) }, void 0, false, {
          lineNumber: 46,
          columnNumber: 11
        }, this)
      ] }, void 0, true, {
        lineNumber: 33,
        columnNumber: 9
      }, this)
    ] }, void 0, true, {
      lineNumber: 15,
      columnNumber: 7
    }, this) }, void 0, false, {
      lineNumber: 14,
      columnNumber: 5
    }, this)
  );
}
const resources = [
  {
    href: "https://remix.run/start/quickstart",
    text: "Quick Start (5 min)",
    icon: /* @__PURE__ */ jsxDEV(
      "svg",
      {
        xmlns: "http://www.w3.org/2000/svg",
        width: "24",
        height: "20",
        viewBox: "0 0 20 20",
        fill: "none",
        className: "stroke-gray-600 group-hover:stroke-current dark:stroke-gray-300",
        children: /* @__PURE__ */ jsxDEV(
          "path",
          {
            d: "M8.51851 12.0741L7.92592 18L15.6296 9.7037L11.4815 7.33333L12.0741 2L4.37036 10.2963L8.51851 12.0741Z",
            strokeWidth: "1.5",
            strokeLinecap: "round",
            strokeLinejoin: "round"
          },
          void 0,
          false,
          {
            lineNumber: 80,
            columnNumber: 9
          },
          this
        )
      },
      void 0,
      false,
      {
        lineNumber: 72,
        columnNumber: 7
      },
      this
    )
  },
  {
    href: "https://remix.run/start/tutorial",
    text: "Tutorial (30 min)",
    icon: /* @__PURE__ */ jsxDEV(
      "svg",
      {
        xmlns: "http://www.w3.org/2000/svg",
        width: "24",
        height: "20",
        viewBox: "0 0 20 20",
        fill: "none",
        className: "stroke-gray-600 group-hover:stroke-current dark:stroke-gray-300",
        children: /* @__PURE__ */ jsxDEV(
          "path",
          {
            d: "M4.561 12.749L3.15503 14.1549M3.00811 8.99944H1.01978M3.15503 3.84489L4.561 5.2508M8.3107 1.70923L8.3107 3.69749M13.4655 3.84489L12.0595 5.2508M18.1868 17.0974L16.635 18.6491C16.4636 18.8205 16.1858 18.8205 16.0144 18.6491L13.568 16.2028C13.383 16.0178 13.0784 16.0347 12.915 16.239L11.2697 18.2956C11.047 18.5739 10.6029 18.4847 10.505 18.142L7.85215 8.85711C7.75756 8.52603 8.06365 8.21994 8.39472 8.31453L17.6796 10.9673C18.0223 11.0653 18.1115 11.5094 17.8332 11.7321L15.7766 13.3773C15.5723 13.5408 15.5554 13.8454 15.7404 14.0304L18.1868 16.4767C18.3582 16.6481 18.3582 16.926 18.1868 17.0974Z",
            strokeWidth: "1.5",
            strokeLinecap: "round",
            strokeLinejoin: "round"
          },
          void 0,
          false,
          {
            lineNumber: 101,
            columnNumber: 9
          },
          this
        )
      },
      void 0,
      false,
      {
        lineNumber: 93,
        columnNumber: 7
      },
      this
    )
  },
  {
    href: "https://remix.run/docs",
    text: "Remix Docs",
    icon: /* @__PURE__ */ jsxDEV(
      "svg",
      {
        xmlns: "http://www.w3.org/2000/svg",
        width: "24",
        height: "20",
        viewBox: "0 0 20 20",
        fill: "none",
        className: "stroke-gray-600 group-hover:stroke-current dark:stroke-gray-300",
        children: /* @__PURE__ */ jsxDEV(
          "path",
          {
            d: "M9.99981 10.0751V9.99992M17.4688 17.4688C15.889 19.0485 11.2645 16.9853 7.13958 12.8604C3.01467 8.73546 0.951405 4.11091 2.53116 2.53116C4.11091 0.951405 8.73546 3.01467 12.8604 7.13958C16.9853 11.2645 19.0485 15.889 17.4688 17.4688ZM2.53132 17.4688C0.951566 15.8891 3.01483 11.2645 7.13974 7.13963C11.2647 3.01471 15.8892 0.951453 17.469 2.53121C19.0487 4.11096 16.9854 8.73551 12.8605 12.8604C8.73562 16.9853 4.11107 19.0486 2.53132 17.4688Z",
            strokeWidth: "1.5",
            strokeLinecap: "round"
          },
          void 0,
          false,
          {
            lineNumber: 122,
            columnNumber: 9
          },
          this
        )
      },
      void 0,
      false,
      {
        lineNumber: 114,
        columnNumber: 7
      },
      this
    )
  },
  {
    href: "https://rmx.as/discord",
    text: "Join Discord",
    icon: /* @__PURE__ */ jsxDEV(
      "svg",
      {
        xmlns: "http://www.w3.org/2000/svg",
        width: "24",
        height: "20",
        viewBox: "0 0 24 20",
        fill: "none",
        className: "stroke-gray-600 group-hover:stroke-current dark:stroke-gray-300",
        children: /* @__PURE__ */ jsxDEV(
          "path",
          {
            d: "M15.0686 1.25995L14.5477 1.17423L14.2913 1.63578C14.1754 1.84439 14.0545 2.08275 13.9422 2.31963C12.6461 2.16488 11.3406 2.16505 10.0445 2.32014C9.92822 2.08178 9.80478 1.84975 9.67412 1.62413L9.41449 1.17584L8.90333 1.25995C7.33547 1.51794 5.80717 1.99419 4.37748 2.66939L4.19 2.75793L4.07461 2.93019C1.23864 7.16437 0.46302 11.3053 0.838165 15.3924L0.868838 15.7266L1.13844 15.9264C2.81818 17.1714 4.68053 18.1233 6.68582 18.719L7.18892 18.8684L7.50166 18.4469C7.96179 17.8268 8.36504 17.1824 8.709 16.4944L8.71099 16.4904C10.8645 17.0471 13.128 17.0485 15.2821 16.4947C15.6261 17.1826 16.0293 17.8269 16.4892 18.4469L16.805 18.8725L17.3116 18.717C19.3056 18.105 21.1876 17.1751 22.8559 15.9238L23.1224 15.724L23.1528 15.3923C23.5873 10.6524 22.3579 6.53306 19.8947 2.90714L19.7759 2.73227L19.5833 2.64518C18.1437 1.99439 16.6386 1.51826 15.0686 1.25995ZM16.6074 10.7755L16.6074 10.7756C16.5934 11.6409 16.0212 12.1444 15.4783 12.1444C14.9297 12.1444 14.3493 11.6173 14.3493 10.7877C14.3493 9.94885 14.9378 9.41192 15.4783 9.41192C16.0471 9.41192 16.6209 9.93851 16.6074 10.7755ZM8.49373 12.1444C7.94513 12.1444 7.36471 11.6173 7.36471 10.7877C7.36471 9.94885 7.95323 9.41192 8.49373 9.41192C9.06038 9.41192 9.63892 9.93712 9.6417 10.7815C9.62517 11.6239 9.05462 12.1444 8.49373 12.1444Z",
            strokeWidth: "1.5"
          },
          void 0,
          false,
          {
            lineNumber: 142,
            columnNumber: 9
          },
          this
        )
      },
      void 0,
      false,
      {
        lineNumber: 134,
        columnNumber: 7
      },
      this
    )
  }
];
"
`;
